integrations/watsonx/src/haystack_integrations/components/generators/watsonx/chat/chat_generator.py

```diff
@@ -0,0 +1,391 @@
+# SPDX-FileCopyrightText: 2025-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import asyncio
+import json
+import os
+from datetime import datetime, timezone
+from typing import Any
+
+from haystack import component, default_from_dict, default_to_dict, logging
+from haystack.dataclasses import (
+    ChatMessage,
+    StreamingCallbackT,
+    StreamingChunk,
+    ToolCall,
+)
+from haystack.utils import Secret, deserialize_secrets_inplace
+from ibm_watsonx_ai import Credentials
+from ibm_watsonx_ai.foundation_models import ModelInference
+
+logger = logging.getLogger(__name__)
+
+
+@component
+class WatsonxChatGenerator:
+    """
+    Enables chat completions using IBM's watsonx.ai foundation models.
+
+    This component interacts with IBM's watsonx.ai platform to generate chat responses
+    using various foundation models. It supports the [ChatMessage](https://docs.haystack.deepset.ai/docs/chatmessage)
+    format for both input and output.
+
+    The generator works with IBM's foundation models including:
+    - granite-13b-chat-v2
+    - llama-2-70b-chat
+    - llama-3-70b-instruct
+    - Other watsonx.ai chat models
+
+    You can customize the generation behavior by passing parameters to the
+    watsonx.ai API through the `generation_kwargs` argument. These parameters
+    are passed directly to the watsonx.ai inference endpoint.
+
+    For details on watsonx.ai API parameters, see
+    [IBM watsonx.ai documentation](https://dataplatform.cloud.ibm.com/docs/content/wsj/analyze-data/fm-parameters.html).
+
+    ### Usage example
+
+    ```python
+    from haystack_integrations.components.generators.watsonx.chat.chat_generator import (
+        WatsonxChatGenerator,
+    )
+    from haystack.dataclasses import ChatMessage
+
+    messages = [ChatMessage.from_user("Explain quantum computing in simple terms")]
+
+    client = WatsonxChatGenerator(
+        api_key=Secret.from_env_var("WATSONX_API_KEY"),
+        model="ibm/granite-13b-chat-v2",
+        project_id=Secret.from_env_var("WATSONX_PROJECT_ID"),
+    )
+    response = client.run(messages)
+    print(response)
+    ```
+    Output:
+    ```
+    {'replies':
+        [ChatMessage(_role=<ChatRole.ASSISTANT: 'assistant'>, _content=
+        [TextContent(text="Quantum computing uses quantum-mechanical phenomena like ....")],
+         _name=None,
+         _meta={'model': 'ibm/granite-13b-chat-v2', 'project_id': 'your-project-id',
+         'usage': {'prompt_tokens': 12, 'completion_tokens': 45, 'total_tokens': 57}})
+        ]
+    }
+    ```
+
+    The component also supports streaming responses and function calling through
+    watsonx.ai's tools parameter.
+    """
+
+    def __init__(
+        self,
+        *,
+        api_key: Secret | None = None,
+        model: str = "ibm/granite-13b-chat-v2",
+        project_id: Secret | None = None,
+        api_base_url: str | None = None,
```

Comment:
For consistency with the embedders could we update this to 
```suggestion
        api_base_url: str = "https://us-south.ml.cloud.ibm.com",
```